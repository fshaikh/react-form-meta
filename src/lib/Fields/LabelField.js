import React from 'react';
import PropTypes from 'prop-types';

export default function LabelField(props) {
    const field = props.field;
    const requiredUI = field.required ? <span className="required">*</span> : '';
    const showLabel = field.showLabel === undefined ? true : field.showLabel;

    if(showLabel) {
        return (
            <label className={props.className} htmlFor={field.name}>
                {field.label} 
                {requiredUI}
            </label> 
        );
    }
    return '';
}

LabelField.propTypes = {
    field: PropTypes.object.isRequired,
    className: PropTypes.string
}