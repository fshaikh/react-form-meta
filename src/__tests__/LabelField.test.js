import React from 'react';
import LabelField from '../lib/Fields/LabelField';
import renderer from 'react-test-renderer';

describe('Label Field',() => {
    describe('Component tests for Label Field',() => {
        it('should check for prop types',() => {
            const field = {
                required: true,
                name: 'firstName',
                label:'First Name'
            };
            const tree = renderer.create(<LabelField field={field} className={[]}/>);
        });
    });

    describe('Snapshot tests for Label Field', () => {
        it('should render correctly for required field', () => {
            const field = {
                required: true,
                name: 'firstName',
                label:'First Name'
            };
            const tree = renderer.create(<LabelField field={field} />).toJSON();
            expect(tree).toMatchSnapshot();
        });
    
        it('should render correctly for non-required field', () => {
            const field = {
                required: false,
                name: 'firstName',
                label:'First Name'
            };
            const tree = renderer.create(<LabelField field={field} />).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
   
});