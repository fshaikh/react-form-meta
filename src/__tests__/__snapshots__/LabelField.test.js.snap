// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Field Snapshot tests for Label Field should not render when showLabel is false 1`] = `
<label
  className={undefined}
  htmlFor="firstName"
>
  First Name
  
</label>
`;

exports[`Label Field Snapshot tests for Label Field should render correctly for non-required field 1`] = `
<label
  className={undefined}
  htmlFor="firstName"
>
  First Name
  
</label>
`;

exports[`Label Field Snapshot tests for Label Field should render correctly for required field 1`] = `
<label
  className={undefined}
  htmlFor="firstName"
>
  First Name
  <span
    className="required"
  >
    *
  </span>
</label>
`;
